cmd/utils/flags.go

```diff
@@ -1726,6 +1732,38 @@ func SetEthConfig(ctx *cli.Context, stack *node.Node, cfg *eth.Config) {
 		cfg.NetworkId = *cfg.Genesis.GetNetworkID()
 	}
 
+	if ctx.GlobalIsSet(EthProtocolsFlag.Name) {
+		protocolVersions := SplitAndTrim(ctx.GlobalString(EthProtocolsFlag.Name))
+		if len(protocolVersions) > 0 {
+			for _, versionString := range protocolVersions {
+				version, err := strconv.ParseUint(versionString, 10, 0)
+				if err != nil {
+					continue
+				}
+
+				isValid := false
+				for _, proto := range eth.DefaultProtocolVersions {
+					if proto == uint(version) {
+						isValid = true
+					}
+				}
+
+				if !isValid {
+					Fatalf("--%s must be one of %s", EthProtocolsFlag.Name, strings.Join(strings.Fields(fmt.Sprint(eth.DefaultProtocolVersions)), ","))
+				}
+				cfg.ProtocolVersions = append(cfg.ProtocolVersions, uint(version))
+			}
+		}
+
+		// sort protocol version desceding
+		sort.Slice(cfg.ProtocolVersions, func(i, j int) bool { return cfg.ProtocolVersions[i] > cfg.ProtocolVersions[j] })
```

Comment:
> // Protocol versions are the supported versions of the eth protocol (first is primary)
> https://github.com/etclabscore/core-geth/pull/244/files#diff-cb10e99654aa32525bbb358ea722f79e26da92b3ecdc70e29ce1d711f1be3de4R116

Do we really want to sort these?

Reply:
I found a place in the network layer where on the server side logic they peek the highest one. But yeah, let's better not break the primary logic here by just a consumption. Fixing

Reply:
Interesting. Is "a place in the network layer where on the server side logic they peek the highest one" worth spinning off into own investigate/fix issue?

Reply:
No, as the logic there was in a different mindset. Will try to find this again and let you know in chat. Let's merge this one once ready